firestore/FirestoreSwiftUIExample/ViewModels/RestaurantListViewModel.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  RestaurantListViewModel.swift
+//  FirestoreSwiftUIExample
+//
+
+import Combine
+import Firebase
+
+class RestaurantListViewModel: ObservableObject {
+
+    @Published var restaurants: [Restaurant] = []
+    private var db = Firestore.firestore()
+
+    func fetchData() {
+        db.collection("restaurants").addSnapshotListener { (querySnapshot, error) in
+            if let error = error {
+                print("Error getting restaurants: \(error.localizedDescription)")
+                return
+            }
+
+            self.restaurants = querySnapshot?.documents.compactMap { document in
+                try? document.data(as: Restaurant.self)
```

Comment:
Consider logging the error instead of discarding it. Something like
```swift
do {
  return try document.data(as: Restaurant.self)
} catch let error {
  print(error)
  return nil
}
```